packages/sync/src/modules/Full_Sync.php

```diff
@@ -700,4 +764,57 @@ private function read_option( $name, $default = null ) {
 
 		return $value;
 	}
+
+	/**
+	 * Prefix of the blog lock transient.
+	 *
+	 * @access public
+	 *
+	 * @var string
+	 */
+	const ENQUEUE_LOCK_TRANSIENT_PREFIX = 'jp_sync_enqueue_lock_';
+	/**
+	 * Lifetime of the blog lock transient.
+	 *
+	 * @access public
+	 *
+	 * @var int
+	 */
+	const ENQUEUE_LOCK_TRANSIENT_EXPIRY = 15; // Seconds.
+
+	/**
+	 * Attempt to lock enqueueing when the server receives concurrent requests from the same blog.
+	 *
+	 * @access public
+	 *
+	 * @param int $expiry  enqueue transient lifetime.
+	 * @return boolean True if succeeded, false otherwise.
+	 */
+	public function attempt_enqueue_lock( $expiry = self::ENQUEUE_LOCK_TRANSIENT_EXPIRY ) {
+		$transient_name = $this->get_concurrent_enqueue_transient_name();
+		$locked_time    = get_site_transient( $transient_name );
+		if ( $locked_time ) {
+			return false;
+		}
+		set_site_transient( $transient_name, microtime( true ), $expiry );
```

Comment:
Any good reason to use a network-wide transient vs a blog-specific transient? If we have two independent subsites enqueueing items in the same time, why would we want to block both of them?

Reply:
See: https://github.com/Automattic/jetpack/pull/13954/#discussion_r342559700